File: model/event.go
package model

import "time"

type Event struct {
	Id         string         `json:"id"`
	EventType  string         `json:"event_type"`
	UserId     string         `json:"user_id"`
	Timestamp  time.Time      `json:"timestamp"`
	Properties map[string]any `json:"properties"`
}

File: frontend-react/src/model/event.ts
export interface AnalyticsEvent {
    id: string;
    timestamp: Date;
    eventType: string;
    userId: string;
    // biome-ignore lint/suspicious/noExplicitAny: <explanation>
    properties: Record<string, any>;
}

export interface RawEventRow {
    id: string;
    event_type: string;
    user_id: string;
    timestamp: bigint; // DuckDB returns bigint for timestamp
    properties: string;

    // biome-ignore lint/suspicious/noExplicitAny: <explanation>
    [key: string]: any;
}
